openquake/mbt/tools/fault_modeler/README.md

```diff
@@ -0,0 +1,170 @@
+
+build_fault_model(geojson_file='FaultDatabase.geojson', 
+                  defaults=defaults)
+```
+
+As final options, optional parameters can be specified in an external configuration file (see below).
+
+However, providing an optional setting will always override the default (hardcoded) value for all the faults in the database at the same time. To provide fault specific parametrisation, it is possible to:
+
+1. Manually modify the geojson by adding the key-value pair to the specific fault item
+
+2. Use the following programmatic approach:
+
+```python
+from openquake.mbt.tools.fault_modeler.fault_source_modeler import fault_database
+from openquake.mbt.tools.fault_modeler.fault_source_modeler import build_model_from_db
+
+# Import the database
+fault_db = fault_database()
+fault_db.import_from_geojson(geojson_file)
+
+# Add a key/value to fault with id 1
+fault_db.add_property('upper_seismogenic_depth', value=20, id=1)
+
+# Create and export the model
+build_model_from_db(fault_db, xml_output='FaultModel.xml')
+```
+(note that the function *build_model_from_db* shares the same options of *build_fault_model*)
+
+The *fault_database* class can be used also to export the modified database to geojson format:
+
+```python
+fault_db.export_geojson(geojson_file)
+```
+
+## Using non-standard Geojson file formats
+
+In case a geojson database with *non-standard* keyword (e.g. an old version),
+it is possible to specify the translation map with a simple dictionary:
+
+```
+param_map = {'source_id': 'ogc_fid',
+             'name': 'ns_name',
+             'average_dip': 'ns_average_dip',
+             'average_rake': 'ns_average_rake',
+             'net_slip_rate': 'ns_net_slip_rate',
+             'vert_slip_rate': 'ns_vert_slip_rate',
+             'strike_slip_rate': 'ns_strike_slip_rate',
+             'shortening_rate': 'ns_shortening_rate',
+             'dip_dir': 'ns_dip_dir',
+             'dip_slip_rate': 'ns_dip_slip_rate'}
+
+build_fault_model(geojson_file='FaultDatabase.geojson',
+                  xml_output='FaultModel.xml')
+                  param_map=param_map)
+```
+
+As well, the translation information can be specified in the configuration file (see following section)
+
+## Configuration File Format
+
+The configuration file (.ini) replaces the need of manually passing the arguments to the main function.
+The file has the following standard format:
+
+```
+[config]
+geojson_file = FaultDatabase.geojson
+xml_output = FaultModel.xml
+select_list = 1,2,3
+
+[param_map]
+source_id = ogc_fid
+name = ns_name
+average_dip = ns_average_dip
+average_rake = ns_average_rake
+net_slip_rate = ns_net_slip_rate
+vert_slip_rate = ns_vert_slip_rate
+strike_slip_rate = ns_strike_slip_rate
+shortening_rate = ns_shortening_rate
+dip_dir = ns_dip_dir
+dip_slip_rate = ns_dip_slip_rate
+
+[defaults]
+upper_seismogenic_depth = 0
+lower_seismogenic_depth = 15
+tectonic_region_type = Active Shallow Crust
+rupture_mesh_spacing = 5
```

Comment:
The description of ini file is not so clear: I suggest the same syntax you can found in a man output:
```
# constant string as plain text
geojson_file = 

# a mandatory parameter as description included in <>
geojson_file = <geojson_filename>

# an optional value as description included in []
geojson_file = [geojson_filename]

# a list of alternatively choices is a sequence of these options separated by pipe (|)
option = <option_A|option_B|option_C>

# an open list of parameter is described as nested [] blocks:
option_list = <first_element [, second_element [, ...]]>
```

Reply:
I'll open an issue to fix this later (and improve the whole README).